### PR TITLE
lua_api.txt: Refine 'Item types' for 4e2eeee

### PR DESCRIPTION
**Pull request by SmallJoker**
@paramat

### CODE REVIEW

**Comment by paramat** [comment]
:+1:

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1365,16 +1365,15 @@ Item types
...
 
-* Node: An item that can be placed in the world as a node.
-* Tool: An item that can be worn out and repaired, cannot be stacked, and often
-  digs or deals damage.
-* Craftitem: A general purpose item that can be used in crafting recipes, eaten,
-  dropped, or used on things in the world.
+* Node: Can be placed in the world's voxel grid
+* Tool: Has a wear property but cannot be stacked. The default use action is to
+  dig nodes or hit objects according to its tool capabilities.
+* Craftitem: Cannot dig nodes or be placed
```
Comment (ClobberXD) [inline doc/lua_api.txt]: .. "can be used in crafting recipes" ?

File: doc/lua_api.txt [hunk 2]
```diff
@@ -1365,16 +1365,15 @@ Item types
...
 
 Amount and wear
 ---------------
 
-All item stacks have an amount between 0 to 65535. It is 1 by
+All item stacks have an amount between 0 and 65535. It is 1 by
```
Comment (ClobberXD) [inline doc/lua_api.txt]: "... stacks have a wear value of 0 to 65535 ..."

File: doc/lua_api.txt (hunk 1 above)
Comment (paramat) [reply doc/lua_api.txt]: So can nodes.

File: doc/lua_api.txt (hunk 2 above)
Comment (paramat) [reply doc/lua_api.txt]: No, wear is one usage.
Best wording is '... item stacks have a value ...'.

Comment (paramat) [reply doc/lua_api.txt]: Ugh, no i'm wrong. Amount is the number of items in that stack, so is fine.

Comment (ClobberXD) [reply doc/lua_api.txt]: Oh... I thought this was the `wear` section. Sorry for the misunderstanding >.<

**Comment by paramat** [comment]
:+1:

File: doc/lua_api.txt (hunk 1 above)
Comment (ClobberXD) [reply doc/lua_api.txt]: True...

"Cannot be used to interact with the world" maybe?

Comment (SmallJoker) [reply doc/lua_api.txt]: Well, it can be dropped and dropped items activate pressure plates.

**Pull request merged**